### PR TITLE
API View for azfile

### PR DESCRIPTION
**Pull request by souravgupta-msft**
[azfile - API View](https://apiview.dev/Assemblies/Review/5ca66dbadf784665a532335b9d6795e4)

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
The CI failure is due to missing constant for the semantic version (it's not immediately clear from the error message).

**Comment by jhendrixMSFT** [comment]
Also, at some point you'll need to update the root `doc.go` file with some "getting started" content/links.

File: sdk/storage/azfile/service/models.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+func (o *GetPropertiesOptions) format() *generated.ServiceClientGetPropertiesOptions {
+	return nil
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// SetPropertiesOptions provides set of options for Client.SetProperties
+type SetPropertiesOptions struct {
+	// The set of CORS rules.
+	Cors []*ShareCorsRule
```
Comment (tasherif-msft) [inline sdk/storage/azfile/service/models.go:43]: Like blobs and queues, let's make all acronyms capitalized

File: sdk/storage/azfile/share/models.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+// ---------------------------------------------------------------------------------------------------------------------
+
+// GetAccessPolicyOptions contains the optional parameters for the Client.GetAccessPolicy method.
+type GetAccessPolicyOptions struct {
+	// LeaseAccessConditions contains optional parameters to access leased entity.
+	LeaseAccessConditions *LeaseAccessConditions
+}
+
+// SignedIdentifier - Signed identifier.
+type SignedIdentifier = generated.SignedIdentifier
```
Comment (tasherif-msft) [inline sdk/storage/azfile/share/models.go:98]: `Id` -> `ID` please change all instances of this in the codebase

File: sdk/storage/azfile/internal/generated/autorest.md [hunk 3]
```diff
@@ -123,3 +123,49 @@ directive:
     $.FileContentType["x-ms-parameter-grouping"].name = "share-file-http-headers";
```
Comment (tasherif-msft) [inline sdk/storage/azfile/internal/generated/autorest.md:123]: For acronyms on the generated layer you'll need to use transforms, please refer to blobs package

File: sdk/storage/azfile/service/models.go (hunk 1 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/service/models.go]: Done

File: sdk/storage/azfile/share/models.go (hunk 2 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/share/models.go:98]: All instances of `ID` are already capitalized.

File: sdk/storage/azfile/internal/generated/autorest.md (hunk 3 above)
Comment (souravgupta-msft) [reply sdk/storage/azfile/internal/generated/autorest.md:123]: This is also an autorest transform. Can you share what exactly you are referring to?

**Review by tasherif-msft — APPROVED** [review]

**Review by vibhansa-msft — APPROVED** [review]

File: sdk/storage/azfile/file/client.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+//   - cred - a SharedKeyCredential created with the matching file's storage account and access key
+//   - options - client options; pass nil to accept the default values
+func NewClientWithSharedKeyCredential(fileURL string, cred *SharedKeyCredential, options *ClientOptions) (*Client, error) {
+	return nil, nil
+}
+
+// NewClientFromConnectionString creates an instance of Client with the specified values.
+//   - connectionString - a connection string for the desired storage account
+//   - shareName - the name of the share within the storage account
+//   - directoryName - the name of the directory within the storage account
```
Comment (richardpark-msft) [inline sdk/storage/azfile/file/client.go:53]: Is `directoryName` here a sub-directory? So should the comment be more like "the name of the directory within the share"?

**Pull request merged**